oauth-function-template/handlers/authorizationCode.js

```diff
@@ -35,12 +36,16 @@ const refreshHandler = (req, res, ctx) => {
             }
             return res.status(500).send(packageError(result));
           }
-          const url = provider.innerAuthorizationUrl;
+          const url = (provider.innerAuthorization && provider.innerAuthorization.url) || provider.innerAuthorizationUrl;
           if (url && !!result.data.accessToken) {
               return getInnerAuthorizationResponse(url, result.data.accessToken)
                   .then((isOkStatus) => {
                       if (!isOkStatus) {
-                          accessTokensTable.editRow(storageTokenInfo.externalId, emptyToken);
+                          const eraseCredentials = _.has(provider, 'innerAuthorization.eraseCredentials') ? provider.innerAuthorization.eraseCredentials : true;
```

Comment:
I think you can change it to: 
```suggestion
                          const eraseCredentials = _.get(provider, 'innerAuthorization.eraseCredentials', true);
```

or something like that.